<div dir="rtl" style="text-align: right;">
<h1>بررسی راه‌حل: خواندن مجموعه داده‌های Auto MPG</h1>
</div>

In [1]:
import pandas as pd # calling pandas module

def read_csv():

    # تعریف نام ستون‌ها به عنوان لیست
    names = ["mpg", "cylinders", "displacement", "horsepower", "weight", "acceleration", "model_year", "origin", "car_name"]
    # خواندن فایل csv مربوط به داده
    df = pd.read_csv("statics/data/02_12_auto-mpg.data", header=None, names=names, delim_whitespace=True)
    return df.shape

# فراخوانی تابع و چاپ نتیجه
print(read_csv())

(398, 9)


/tmp/ipykernel_4941/1732141655.py:8: FutureWarning: The 'delim_whitespace' keyword in pd.read_csv is deprecated and will be removed in a future version. Use ``sep='\s+'`` instead
  df = pd.read_csv("statics/data/02_12_auto-mpg.data", header=None, names=names, delim_whitespace=True)


<div dir="rtl" style="text-align: right;">

<p>بر اساس صورت مسئله،  باید مجموعه داده‌های 
Auto MPG
 را به عنوان ورودی دریافت کنیم. در کد بالا، در خط ۱، ما ماژول 
 pandas
  را برای خواندن مجموعه داده‌ها وارد کردیم. سپس ما تابع 
 read_csv()
  را پیاده‌سازی کردیم.</p>
<p>به هدر آن در خط ۳ نگاه کنید. در خط ۶، ما نام ستون‌ها را در یک لیست به نام 
names
 قرار دادیم. همان‌طورز که در صورت مسئله گفته شد نام ستون‌ها را از مستندات به دست می‌آوریم. توجه داشته باشید که 
 names
  حاوی ۹ مقدار رشته‌ای است:</p>
<ul>
<li>mpg</li>
<li>cylinders</li>
<li>displacement</li>
<li>horsepower</li>
<li>weight</li>
<li>acceleration</li>
<li>model_year</li>
<li>origin</li>
<li>car_name</li>
</ul>
<p>توجه داشته باشید که ما فضای خالی را با _ در 
model year
 و 
 car name
  جایگزین کرده‌ایم.</p>
<p>خط ۸ مهم‌ترین خط است. ما از تابع داخلی 
read_csv()
 از پانداس استفاده می‌کنیم. همانطور که آشنا هستید، این تابع برخی آرگومان‌ها را به شرح زیر می‌پذیرد:</p>
<ul>
<li>path: مسیر به مجموعه داده</li>
<li>header: در صورتی که ستون‌ها به صراحت ارائه شده‌اند، به None تنظیم می‌شود.</li>
<li>names: لیستی از نام‌های ستون‌ها که قرار است استفاده شوند</li>
<li>delim_whitespace: مشخص می‌کند که آیا فضای خالی باید به عنوان sep استفاده شود یا نه</li>
</ul>
<p>پس برای 
path
، ما آدرس فایل را می‌دهیم: 
statics/data/02_12_auto-mpg.data
. از آنجا که ما نام‌های ستون‌ها را می‌دانیم و آنها را به صراحت به تابع به عنوان 
names=names
 داده‌ایم، 
 header
  را به عنوان 
  None
   تنظیم کرده‌ایم. کفته شده که میان داده‌ها فضاهای خالی وجود دارند و ما باید آنها را به عنوان 
   sep
    استفاده کنیم، بنابراین 
    delim_whitespace
     به 
     true
      تنظیم شده است. این تابع داخلی یک 
      dataframe df
       را به عنوان نتیجه برمی‌گرداند. در خط ۹، ما شکل، یک ویژگی از 
       dataframe df
        را برمی‌گردانیم.</p>
<p>حالا، به خط ۱۲ نگاه کنید، جایی که ما تابع 
read_csv()
 را فرا می‌خوانیم. این یک دوتایی به شکل (ردیف‌ها، ستون‌ها) را برمی‌گرداند، که در نهایت چاپ می‌شود. برای این مجموعه داده، تابع 
 (398, 9)
  را برمی‌گرداند زیرا ۳۹۸ ردیف و ۹ ستون وجود دارد.</p>
</div>
